src/actions/index.js

```diff
@@ -141,6 +141,51 @@ export function toggleShowFeatureHints(showFeatureHints) {
   };
 }
 
+export const UPDATE_PREFERENCES = 'UPDATE_PREFERENCES';
```

Comment:
I wouldn't add this as a redux action. I would just have a useEffect in settings.modal that would track changes to `setShowDataSetPreview` and make the below call.